### PR TITLE
feat!: Add return types to helper functions.

### PR DESCRIPTION
**Pull request by vxern**
Additional changes:

- Make helper function code more consistent: The variables storing the result of an API endpoint being contacted have all been renamed to 'result' for a single result or 'results' for an array of results.
- Use standard naming techniques: Several files were missing the 'get' keyword. The 'remove' keyword has been replaced with 'delete' to correspond to both the Discord API and the rest of the code.

### CODE REVIEW

File: helpers/channels/deleteChannel.ts [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import type { Bot } from "../../bot.ts";
-import { DiscordChannel } from "../../types/discord.ts";
 
 /** Delete a channel in your server. Bot needs MANAGE_CHANNEL permissions in the server. Bot needs MANAGE_THREADS permissions in the server if deleting thread. */
-export async function deleteChannel(bot: Bot, channelId: bigint, reason?: string) {
-  await bot.rest.runMethod<DiscordChannel>(
+export async function deleteChannel(bot: Bot, channelId: bigint, reason?: string): Promise<void> {
+  return void await bot.rest.runMethod(
```
Comment (Skillz4Killz) [inline helpers/channels/deleteChannel.ts:5]: What is the reasoning to add return void here?

File: helpers/discovery/editDiscovery.ts [hunk 2]
```diff
@@ -16,7 +21,7 @@ export async function editDiscovery(bot: Bot, guildId: bigint, data: ModifyGuild
 
   return {
     guildId,
-    primaryCategoryId: result.primary_category_id,
+    primaryCategoryId: BigInt(result.primary_category_id),
```
Comment (Skillz4Killz) [inline helpers/discovery/editDiscovery.ts:24]: Are u sure this is a snowflake and not a categor name?

File: helpers/discovery/getDiscovery.ts [hunk 3]
```diff
@@ -11,7 +21,7 @@ export async function getDiscovery(bot: Bot, guildId: bigint) {
 
   return {
     guildId,
-    primaryCategoryId: result.primary_category_id,
+    primaryCategoryId: BigInt(result.primary_category_id),
```
Comment (Skillz4Killz) [inline helpers/discovery/getDiscovery.ts:24]: Confirm this is indeed a snowflake and if it is use the transformers.snowflake

File: helpers/emojis/getEmojiUrl.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { Bot } from "../../bot.ts";
 
 /** Creates a url to the emoji from the Discord CDN. */
-export function emojiUrl(bot: Bot, id: bigint, animated = false) {
+export function getEmojiURL(_bot: Bot, id: bigint, animated = false): string {
```
Comment (Skillz4Killz) [inline helpers/emojis/getEmojiUrl.ts:4]: Hmm renaming this is a bit weird as there is no fetching involved here. The get makes me think there was some fetching going on here

File: helpers/guilds/getBan.ts [hunk 5]
```diff
@@ -1,16 +1,22 @@
 import type { Bot } from "../../bot.ts";
+import { User } from "../../transformers/member.ts";
 import { DiscordBan } from "../../types/discord.ts";
 
+export type Ban = {
+  reason: string | undefined;
```
Comment (Skillz4Killz) [inline helpers/guilds/getBan.ts:6]:
```suggestion
  reason?: string;
```

File: helpers/guilds/getVanityUrl.ts [hunk 6]
```diff
@@ -1,16 +1,15 @@
...
+export type VanityUrl = {
+  code?: string;
+  uses: number;
+};
 
 /** Returns the code and uses of the vanity url for this server if it is enabled else `code` will be null. Requires the `MANAGE_GUILD` permission. */
-export async function getVanityUrl(bot: Bot, guildId: bigint) {
-  const result = await bot.rest.runMethod<Partial<DiscordInviteMetadata>>(
+export async function getVanityUrl(bot: Bot, guildId: bigint): Promise<VanityUrl> {
+  return bot.rest.runMethod<VanityUrl>(
```
Comment (Skillz4Killz) [inline helpers/guilds/getVanityUrl.ts:10]:
```suggestion
  return await bot.rest.runMethod<VanityUrl>(
```

File: helpers/integrations/getIntegrations.ts [hunk 7]
```diff
@@ -1,35 +1,19 @@
...
-        expire_grace_period: res.expire_grace_period,
-        user: res.user,
-        account: res.account,
-        synced_at: res.synced_at,
-        subscriber_count: res.subscriber_count,
-        revoked: res.revoked,
-        application: res.application,
-      });
+    results.map((result) => {
+      const integration = bot.transformers.integration(bot, { guild_id: guildId.toString(), ...result });
```
Comment (Skillz4Killz) [inline helpers/integrations/getIntegrations.ts:16]: Undo this change, we do not use spread operator in dd, please write them manually

File: helpers/interactions/commands/editInteractionResponse.ts [hunk 8]
```diff
@@ -101,7 +103,7 @@ export async function editInteractionResponse(
   );
 
   // If the original message was edited, this will not return a message
-  if (!options.messageId) return result as undefined;
+  if (!options.messageId) return undefined;
```
Comment (Skillz4Killz) [inline helpers/interactions/commands/editInteractionResponse.ts:106]:
```suggestion
  if (!options.messageId) return;
```

File: helpers/interactions/sendInteractionResponse.ts [hunk 9]
```diff
@@ -36,7 +36,7 @@ export async function sendInteractionResponse(
 
   // A reply has never been send
   if (bot.cache.unrepliedInteractions.delete(id)) {
-    return await bot.rest.sendRequest<undefined>(bot.rest, {
+    return <undefined> await bot.rest.sendRequest(bot.rest, {
```
Comment (Skillz4Killz) [inline helpers/interactions/sendInteractionResponse.ts:39]: why?

File: helpers/members/editBotNickname.ts [hunk 10]
```diff
@@ -1,15 +1,19 @@
...
+): Promise<string | undefined> {
+  const result = await bot.rest.runMethod<{ nick?: string }>(
     bot.rest,
     "PATCH",
     bot.constants.routes.USER_NICK(guildId),
     options,
   );
 
-  if (!response?.nick) return;
+  if (!result?.nick) return undefined;
```
Comment (Skillz4Killz) [inline helpers/members/editBotNickname.ts:16]:
```suggestion
  if (!result?.nick) return;
```

File: helpers/webhooks/sendWebhookMessage.ts [hunk 11]
```diff
@@ -32,7 +38,7 @@ export async function sendWebhook(bot: Bot, webhookId: bigint, webhookToken: str
       components: options.components?.map((component) => bot.transformers.reverse.component(bot, component)),
     },
   );
-  if (!options.wait) return;
+  if (!options.wait) return undefined;
```
Comment (Skillz4Killz) [inline helpers/webhooks/sendWebhookMessage.ts:41]:
```suggestion
  if (!options.wait) return;
```

File: helpers/channels/deleteChannel.ts (hunk 1 above)
Comment (vxern) [reply helpers/channels/deleteChannel.ts:5]: It is a personal preference. The code works just fine without it, but I feel that it makes the code more consistent since a value is always returned, even if its void.

File: helpers/discovery/editDiscovery.ts (hunk 2 above)
Comment (vxern) [reply helpers/discovery/editDiscovery.ts:24]: I do not know. These endpoints are not documented.

File: helpers/discovery/getDiscovery.ts (hunk 3 above)
Comment (vxern) [reply helpers/discovery/getDiscovery.ts:24]: On it.

Comment (vxern) [reply helpers/discovery/getDiscovery.ts:24]: After some digging: It is not a snowflake. Just a number. Fixed in latest commit.

File: helpers/interactions/sendInteractionResponse.ts (hunk 9 above)
Comment (vxern) [reply helpers/interactions/sendInteractionResponse.ts:39]: The generic annotation in `bot.rest.sendRequest()` is redundant, since the value is irrelevant, however the function requires that an `undefined` be returned if there is no value, so cast Promise return type to `undefined`.

File: helpers/webhooks/sendWebhookMessage.ts (hunk 11 above)
Comment (vxern) [reply helpers/webhooks/sendWebhookMessage.ts]: Updated in latest commit.

File: helpers/members/editBotNickname.ts (hunk 10 above)
Comment (vxern) [reply helpers/members/editBotNickname.ts]: Updated in latest commit.

File: helpers/interactions/commands/editInteractionResponse.ts (hunk 8 above)
Comment (vxern) [reply helpers/interactions/commands/editInteractionResponse.ts]: Updated in latest commit.

File: helpers/emojis/getEmojiUrl.ts (hunk 4 above)
Comment (vxern) [reply helpers/emojis/getEmojiUrl.ts:4]: I thought that adding the get keyword to the file/function names that were missing them would make them more consistent.

File: helpers/discovery/editDiscovery.ts (hunk 2 above)
Comment (vxern) [reply helpers/discovery/editDiscovery.ts:24]: Updated type.

File: helpers/channels/deleteChannel.ts (hunk 1 above)
Comment (Skillz4Killz) [reply helpers/channels/deleteChannel.ts:5]: In that case id prefer return await bot.rest.runMethod<void>()

File: helpers/emojis/getEmojiUrl.ts (hunk 4 above)
Comment (Skillz4Killz) [reply helpers/emojis/getEmojiUrl.ts:4]: Consistency is good but not if it causes confusion. `get` should be reserved for prefixing on functions that make a request to discord. Otherwise its just a util function and imo something like this can in fact be removed from `helpers` and moved to utils folder

File: helpers/channels/deleteChannel.ts (hunk 1 above)
Comment (vxern) [reply helpers/channels/deleteChannel.ts:5]: Sure, that works too. Going to update.

File: helpers/interactions/sendInteractionResponse.ts (hunk 9 above)
Comment (Skillz4Killz) [reply helpers/interactions/sendInteractionResponse.ts:39]: I'm not a fan of this style. I prefer the older way its much better on the eyes and readability. the generic undefined passed to sendRequest tells typescript that it is returning undefined which is exactly what we need imo. No?

Comment (vxern) [reply helpers/interactions/sendInteractionResponse.ts:39]: Indeed, and yes, perhaps it is indeed better for readability. I'll update the code.

File: helpers/guilds/getBan.ts (hunk 5 above)
Comment (vxern) [reply helpers/guilds/getBan.ts:6]: I believe I used `undefined` here because the TS analyser had issues with inferring the type in `getBans.ts`. Updated in next commit.

File: helpers/integrations/getIntegrations.ts (hunk 7 above)
Comment (vxern) [reply helpers/integrations/getIntegrations.ts]: Done.

**Review by Skillz4Killz — APPROVED** [review]
